### ISSUE TITLE
Alternatives, since this uBO port is pretty much dead and unusable for many?

### ISSUE THREAD

**Issue by amarendra** [el1t/uBlock-Safari#145]
What other extensions or apps we can use to check tracking on Safari, other than switching browsers? Let's discuss this here since a lot of users would come here first.

For me this extension stopped working after last Safari update. I tried reinstalling and some other tweaks but I still have the same issue. The uBlock Safari on browser bar is just a blank space for example: 
<img width="107" alt="screenshot 2018-11-14 09 28 33" src="https://user-images.githubusercontent.com/79829/48459389-cb5efc80-e7ef-11e8-918e-17feacfacdf4.png">

However I personally would like to move on to some other tracking protection or content blocker.

#130 #143 #142

**Comment by noof2** [comment]
I'm using Better on iOS and macOS.  
https://better.fyi
It is lightweight.  The extension does not need permission to read or transmit Web content which I like from a privacy and security standpoint.  (The extension can't see your credit card number for instance.)  The developers are responsive and seem committed to the project.

**Comment by nbara** [comment]
Better is fine. I purchased it last week, but I must say it clearly does not do as well as uBlock origin in terms of performance. It's filters are not customisable, nor is is possible to block (or unblock) specific content on a page.

**Comment by amarendra** [comment]
@nbara Not able to blocking specific content is fine for me at least. But does it block tracking completely? Also, it's source code is open but I notice there is just one developer on their own Gitlab instance.

**Comment by MysticalOS** [comment]
I had tried "Better" as well and stopped using it within 10 min

It Broke some websites right off the bat and not being able to whitelist/fix made it unusuable. It did NOT block things I actually did want to block. Things that would automatically blocked by easylist filters, so that tells me better isn't even using already WELL established filters that are out there.

to be honest that's why most adblockers out there that aren't ublock, fail. Everyone wants their own filter when there are already well established and vetted filter lists. Just give a blocker that does what ublock does. Provides an efficient engine that lets us pick and choose from the already well established filters of the internet. I don't care how good you think "YOUR" filter is. don't make me choose only YOUR filter. :)

My 2 cents. Ublock still works "well enough" to still beat competition even being partially broken. That says a lot

**Comment by amarendra** [comment]
@MysticalOS I have really been struggling been uBlock of late and I really do not want to start looking for alternatives the day it might stop working completely or is really out of date that it becomes useless. 

I have asked Better devs to chip in here if possible (https://twitter.com/amarendra/status/1062902092044558337). Even though the project is not on GitHub they may have accounts. But to be honest to me too it seems like "my own little side project" in execution.

**Comment by noof2** [comment]
For what it's worth, you actually can White List sites in Better.  The feature is called "Do not block...".  Sites added to the "Do not block" list are automatically synchronized between a user's MacOS and iOS Better installations via iCloud.  I think of Better as more of a privacy tool than an outright ad blocker.  I group it with Privacy Badger which I use on Chrome.  Many ads are blocked, but not all.  I have been happy with Better.

Other ad blockers that seem to be Mojave and Safari 12 friendly include "Adblock for Safari" (icon is stop sign with a hand) and "AdGuard for Safari" both in the Mac App Store.  These aren't uBO either but the original poster was looking to discuss options...

**Comment by MysticalOS** [comment]
"Do not block" is not the same as "don't break disquss on this page that also has 40 ads/auto play videos you do still need to block"

**Comment by tonilampela** [comment]
1Block has been pretty good on iOS so I assume it wouldn't be any worse on macOS either 
https://itunes.apple.com/fi/app/1blocker/id1107421413?l=fi&mt=12

**Comment by MysticalOS** [comment]
It's actually terrible on macOS. I've looked at it. they haven't ported their multiple lists rewrite to macOS yet so it's still a single 50,000 rulec capped list that is poorly updated. 1blocker also uses their own custom list and not widly available updated DAILY public lists. Heck they only seem to update their custom list every once in a blue moon from what I've seen on iOS. I actually use two blockers on iOS. 1blocker cause it has multi list support and covers a lot of bases, and then I use adblock for IOS because (unlike desktop version which is crap) it's simply one list app that downloads latest easylist.

Trust me, I've played with just about every macOS list, they are bad. App store reviews will tell you as much for most of them. We just need El1t to come back or maybe @gorhill to pick up this port. People would happy pay for quality maintenance. I know I would.

**Comment by tonilampela** [comment]
Oh... Didn't realize that. I'm now using Better and while it mostly works fine, it definitely doesn't block quite as well as uBlock did. I'd be also happy to pay would Origin be available on App Store.

**Comment by T100D** [comment]
Using Ad-Guard extention on Safari. Not aware of any problems. Safari 12.01 does not appear to be flawless.

**Comment by freediverx** [comment]
A lot of folks seem to think the problem is that the devs have abandoned uBlock Origin on Safari. The real issue is that Apple is restricting what Safari extensions can do, and starting with the next version of Safari/macOS next year, it will be IMPOSSIBLE to reproduce uBlock Origin’s functionality. Try finding a good ad blocker on iOS… There aren’t any—precisely because of the sort of restrictions that are coming to macOS as well.

We’re fucked when it comes to having a good ad blocker on Safari.

Nothing would make me happier than to be proven wrong.

**Comment by ghost** [comment]
Well to be fair it's both. UBO for Safari could still benefit from development as it is currently still installable/usable. But I guess it's not really worth it to put much energy and time into a fork that will stopp working a year from now (if not sooner) anyway. The future is bleak when it comes to Safari, in general. We will lose the option for a great security/performance tool and there isn't (and will not be) anything like it, by courtesy of the limitations on Apple's side. The solution is, switching to Firefox and live with the impact on battery life.

**Comment by freediverx** [comment]
<sigh>

**Comment by SimonAB** [comment]
I've tried a few (Better, 1block, Magic Lasso, Unicorn) and I keep coming back to **Wipr**. Dead simple (i.e. no customisation) but it works great, doesn't break stuff, and doesn't leave ad placeholders. Same deal on macOS and iOS. Between Wipr and Safari's new anti-tracking system, we have a very good combo imo.

**Comment by Section25** [comment]
I only discovered uBlock ORIGIN fairly recently - until a few months ago I had been using two or three ad blockers at a time to try to manage certain ad saturated websites.

However, a couple of days ago, uBO started misbehaving and ignores my whitelist (a couple of sites I work on and which uBO blocks some WordPress plugins the owners need). The site disable switch is now a universal on/off switch which is a bit tedious.

I keep considering switching to Chrome but every time I try, I hit barriers and head back to Safari.

**Comment by T100D** [comment]
> I only discovered uBlock ORIGIN fairly recently - until a few months ago I had been using two or three ad blockers at a time to try to manage certain ad saturated websites.
> 
> However, a couple of days ago, uBO started misbehaving and ignores my whitelist (a couple of sites I work on and which uBO blocks some WordPress plugins the owners need). The site disable switch is now a universal on/off switch which is a bit tedious.
> 
> I keep considering switching to Chrome but every time I try, I hit barriers and head back to Safari.

Try AdGuard I am satisfied with it for the moment and can keep using Safari.

**Comment by Section25** [comment]
That was one of the previous ad blockers I used but it is only any good when used alongside others.

Its installation actually caused some sites to detect it and block me from seeing content when it is installed but not activated - don't ask me how - but it was the inescapable conclusion of using reductive troubleshooting. I sent a message to the developers with the URLs of the sites but I didn't really expect a response.

**Comment by axeII** [comment]
You can use [pihole](https://pi-hole.net) ⚫️ . It's much simpler solution and works for most ads in there. You don't even need to have installed plugin in your browser.

> Works as a ban list for specific domain address so ads are not loaded in your own browser via DNS. 

Meaning pages suppose to load faster and you don't get those annoying message as "Please disable your ad block" etc.  Also you can block easily ads on iOS this way (via wifi, cellular needs vpn)


Anyway here is public pihole server: 185.187.240.11
Status data: https://pihole-uk.freek.ws/admin/

### Setup

On Mac your can set this server as your own DNS server via. Settings -> Network -> Advanced... -> DNS and there just add IP address `185.187.240.11` . They promise not using any logs.
I have setup raspberry pi running own pihole. 

Btw. I got that ip from here -> https://freek.ws/public-pi-hole/

Edit: Pihole is clever and interesting project but it's not perfect **yet** so it's recommended using combo pihole and 'some' adblock.

**Comment by Section25** [comment]
Actually, I feel a bit stupid now because I just remembered that I use AdGuard DNS so I suppose I am using AdGuard in a way and it is a similar approach to pihole.

 I also use GasMask which maintains a comprehensive list of blocked domains in the hosts file.

**Comment by sweetppro** [comment]
AdGuard has a new extension which works well
https://github.com/AdguardTeam/AdGuardForSafari

**Comment by Section25** [comment]
I can't keep up.
I have installed AdGuard for Safari and I'll see how it goes. At least it seems to be able to do what attracted me to uBO in the first place, i.e. ad a filter to block GDPR notices - apart from everywhere else, I was getting them repeatedly practically every page on eBay.

**Comment by Section25** [comment]
There is something in uBO (probably a list) which blocks ads on eBay which AdGuard doesn't. I've been going through the lists but so far I can't block them with AG and after a while, AG started blasting me with the 50,000 rules rule even as I deselected list after list as fast as I could. It seems to have alerted me the first time a long time after the limit was exceeded.

Anyway, for now, I still have to use both.

**Comment by MysticalOS** [comment]
Sounds like adguard is still only using one list. the 50k limit is only PER LIST. you can get around it by multiple lists. it's a convoluted solution that apple really needs to help developers not need, but it is what it is.

there is also a place for using content blocking and a place for using on load blocking/removal like ublock. content blocking is more efficient because apples api prevents more from loading and lets you block things onload blocking doesn't. but it's far less able to whitelist. I'd love to see a really good content blocker that can act as first defense, followed by a on load blocker like ublock to follow up.. but i'm not satisfied with any of content blockers yet, every one i tried broke too much.

**Comment by amarendra** [comment]
I have tried AdGuard but it doesn't tell me what it is blocking on any given page, if it's blocking anything unlike uBO which tells me upfront what it's blocking (am I missing something here?). It just sits there and I hope it's doing it's job. Another issue is I've to use the companion Electron (something I try not to run on my machine continuously) app and I can't just use the browser extension.

**Comment by MysticalOS** [comment]
Content Blocker extensions can't tell you what they are blocking. The hook apples api and all the blocking is done before page load, so no extension can count the elements block (or help you identify the in case you need to unblock/whitelist them).

**Comment by T100D** [comment]
There is a thing called pi-hole https://pi-hole.net what you can use in a local router or separate unit.

**Comment by tonilampela** [comment]
> I've tried a few (Better, 1block, Magic Lasso, Unicorn) and I keep coming back to **Wipr**. Dead simple (i.e. no customisation) but it works great, doesn't break stuff, and doesn't leave ad placeholders. Same deal on macOS and iOS. Between Wipr and Safari's new anti-tracking system, we have a very good combo imo.

Wipr looks actually pretty good. Blocks ads which other blockers in AppStore didn't. I had actually used it back when it was initially released as legacy extension, but switched to uBO a good while ago.

**Comment by MysticalOS** [comment]
Wipr actually is pretty good because it's first content blocker I've seen on macOS that knows how to use split lists to get around 50k limit (because lets be real, you need about 150k to have room. my ublock origin is always around 120k rules).

Wipr uses 3 lists so it does have up to 150k rules (that's why it installs as 3 extensions). That's a good start. It'd get more points if I could see the lists or add custom rules or my own lists to it. Nothing beats ublock origins ability to use as many lists as you want, then merging them (while also eliminating duplicate rules)

**Comment by amarendra** [comment]
> Not aware of any problems

That's what I have been wondering. As I have noticed in these comments Apple didn't leave a way for users to notice where there was a problem. Everything happens in the background. 

Actually if something is getting blocked and the user is not able to interact with a website properly then, unlike uBO where you could unblock individual elements - API calls, in this you'll have to unblock everything pretty much.

**Comment by kmyid** [comment]
I've tried AdGuard for Safari, which is GPLv3, seems to work perfectly.

**Comment by ghost** [comment]
Can anyone chime in on how Wipr compares to Ublock Origin in terms of blocking and privacy?

**Comment by translit** [comment]
> Can anyone chime in on how Wipr compares to Ublock Origin in terms of blocking and privacy?

Like iOS to Arch Linux.

**Comment by axeII** [comment]
>>Can anyone chime in on how Wipr compares to Ublock Origin in terms of blocking and privacy?

>Like iOS to Arch Linux.

That bad _huh..._

**Comment by MysticalOS** [comment]
I used wipr alone for about 2 days before I got to a site that had a lot of ads with wiper that ublock was able to remove with default lists.

Wipr according to their articles uses two lists. Easylist and Easylist Privacy, that's it. so it's basically as good as ublock if it was only using those two lists.

fortunately those two lists ARE the majority of ads. but you miss out on all the other lists.

**Comment by DavidCWGA** [comment]
> AdGuard has a new extension which works well
> https://github.com/AdguardTeam/AdGuardForSafari

This is what I've been looking for. It has customizable blocklists. So you can turn off the AdGuard filters and turn on the EasyList ones. Perfect. Thank you.

**Comment by amarendra** [comment]
> AdGuard has a new extension which works well
> https://github.com/AdguardTeam/AdGuardForSafari

I just dug a bit deeper and I indeed see many lists there which is awesome and since it's OSS I think others can contribute to add more options and filters which are toggles anyway.

My only gripe, still, is I won't know what's being blocked and what's not and if something is not working on a site I will have to do 0/1 AdGuard for that entire site. 

Also, it starts throwing error when I start enabling filters `Too many filters enabled. Safari cannot use more than 50 000 rules.`. Maybe someone from @AdguardTeam like @Stillness-2 could weigh on it whether there's  a way around it. 1Blocker probably has a workaround for that.

**Comment by uBlock-user** [comment]
There's no alternative that can do all that uBO does. So I will say it anyways -- Dump Safari, use Chrome **OR** Firefox on Mac and install uBO there.

**Comment by amarendra** [comment]
I can move to Firefox but its performance on Mac is way below Safari's. Anyway, I guess I will have to compromise - either learn to live with ads and tracking or move to a browser that puts my Mac's fans on panic mode.

**Comment by uBlock-user** [comment]
and what about Chrome ? You didn't mention that.

**Comment by amarendra** [comment]
Oh, I only use Google products that I must use for work and for that there's Chrome installed. For personal usage or browsing it's just Safari.

**Comment by translit** [comment]
> There's no alternative that can do all that uBO does. So I will say it anyways -- Dump Safari, use Chrome **OR** Firefox on Mac and install uBO there.

Nah. Slow, no integration with Keychain, less shiny.

**Comment by freediverx** [comment]
Exactly. It’s fascinating how poorly Android and Windows users understand Mac users and the reasons why we use the platform and its native apps.

**Comment by uBlock-user** [comment]
> It’s fascinating how poorly Android and Windows users understand Mac users and the reasons why

I have a Mac tablet myself. There's nothing to understand, you have limited options either use Chrome/Firefox and get full functionality of uBO back or stick with Safari and keep looking for alternatives which will never measure up to what uBO offers.

**Comment by freediverx** [comment]
Thanks for stating the obvious. What you still don’t get is that the choice of browser is more important to me than the ad blocker.

**Comment by translit** [comment]
For now I've settled on AdGuard. It's not as capable as uBO but it is native, open source, lets me select lists, does cosmetic filtering, and is [taken seriously](https://github.com/gorhill/uBlock/wiki/Debunking-%22uBlock-Origin-is-less-efficient-than-Adguard%22-claims) by gorhill himself.

Still, it's no replacement for uBO. Safari + uBO was like macOS itself: great looks coupled with power.

**Comment by BillyJoeJimBob** [comment]
I've tried the new app store versions of AB, ABP, AdGuard and Wipr and they simply don't have the filters, speed (especially when scrolling) or customizability of their earlier gallery versions, much less UBO.  Also, the old gallery versions of those extensions are apparently now limited in the number of filters that can be used.  JS Blocker 5 in the legacy Safari Extensions gallery is still fully functional (not limited to just 50K filters) and can currently do some of what UBO for Safari did when it was fully functional, just make sure you don't update it from the Safari gallery version (5.2.2) when you click its icon in the Safari browser toolbar.   It comes with the EasyList, EasyPrivacy, EasyMalware and Peter Lowe's List filters enabled and you can add others including uBlock Origin's built-in filters:  

UBO built-in filters:  https://github.com/uBlockOrigin/uAssets/tree/master/filters

My JS Blocker 5 filters:  
<img width="499" alt="screen shot 2018-12-05 at 3 56 41 pm" src="https://user-images.githubusercontent.com/11702508/49552920-56d92400-f8aa-11e8-81c9-61272549bcfc.png">

Legacy Safari Extension Gallery: https://safari-extensions.apple.com/?q=js%20blocker

JS Blocker 5 link:  https://jsblocker.toggleable.com

JS Blocker 5 has a considerable learning curve compared to UBO, when using filters to block ads it leaves place-holders, it can be slow sometimes when scrolling and it doesn't provide for individual filters or cosmetic blocking, but a work around is to keep the current version of UBO (1.16.0) and only use its "My Filters" section.  This avoids the counter issues and lack of filter updating in that version of UBO.  As a bonus, JS Blocker can run user scripts just like Tamper Monkey and has tools for avoiding canvas fingerprinting.  

Using JS Blocker 5 as an adblocker in Safari isn't ideal but the only other solution that works for me in Safari 12+ is to keep using the current UBO and remembering to manually update filters and always open links in new tabs to avoid the counter and dynamic filter issues.  Hopefully some much improved app store solutions will materialize by the time Safari's legacy extensions are shut down for good.

**Comment by darkmastermindz** [comment]
@freediverx @chiffon-rouge It still happens to work for me in Safari 12.0.2 on macOS Mojave version 10.14.2 Beta (18C52a) and it's the best adblocker right now in my option. 

I just added Ublock-Safari to [Code Triage](https://www.codetriage.com/el1t/ublock-safari). I'm hopeful that we can work something out here. 

@el1t Maybe, we could set some goals / enhancements to prepare for the next version of Safari?

**Comment by BillyJoeJimBob** [comment]
UBO 16.0 for Safari is currently the best option for me in Safari 12 but it is only partially functional.  Clicking links in web sites causes the history and counter issues that have been mentioned in other threads (#122 and #144), but opening links in new tabs seems to avoid this (The LinkThing extension can automate this: https://safari-extensions.apple.com/?q=linkthing).  Static filters still work but dynamic filtering is broken in UBO 16.0, at least in my configuration.  

UBO 15.4 worked well enough in Safari 11 even if it is outdated, if that functionality could be restored to the legacy UBO for Safari 12+ it could at least keep the browser usable until Apple ends the legacy extensions, presumably next fall with Safari 13.  Perhaps by then an acceptable solution eventually arises under the new Safari app system.  Wipr's multi-filter approach may be the way although they aren't there yet in terms of browser performance or even minimal configurability.  Maybe a system-wide, multi-browser Mac OS app like that of Cookie 5 (which works great for cookie management) or AdGuard premium would work (I know, on-going service payments, Russians, etc.).  Maybe an ad-blocking DNS service is the least bad way to go, this works for my streaming devices at least.  

see: 
https://www.opendns.com
https://freek.ws/public-pi-hole/  
https://adguard.com/en/adguard-dns/overview.html

A lot of users of Firefox thought the browser had became a ruined Chrome-clone when Firefox 57 came out last year with a new extension system that ended many long-existing and popular extensions.  Since then much of the functionality of those legacy extensions has returned with rewritten or new extensions under the new system.

**Comment by ghost** [comment]
The real missing feature for me from all the alternatives listed here is cosmetic filtering.
I'm using a combination of Wipr and Ublock Origin with only a few lists (my custom lists + fanboy annoying + EasyList for my country) from now, but I think having both might slow down the page rendering a bit.

**Comment by translit** [comment]
> The real missing feature for me from all the alternatives listed here is cosmetic filtering.

AdGuard can do this. And its element selector is better than that of uBO.

**Comment by ghost** [comment]
I tried it but wasn't convinced: it seems the application/process that you must keep open/running to update lists is a bloaty electron app that uses a lot of RAM.

**Comment by DavidCWGA** [comment]
> I tried it but wasn't convinced: it seems the application/process that you must keep open/running to update lists is a bloaty electron app that uses a lot of RAM.

You don't have to keep the application running, only the menu bar icon, which is a separate process.

Right now the whole thing, including the "bloaty" application, is using about 100Mb on my Mac. That's less than BBEdit, or even the Activity Monitor itself from which I am getting these numbers.

**Comment by ghost** [comment]
Thanks for the suggestions, all Adguard process takes 18OMb of RAM on my mac which compared to Wipr's 30Mb looks bloaty, and the electron app does not terminate after closing the settings panel. I killed Adguard and Adguard Helpers in Activity Monitor (leaving Aguard Safari icon running), and content blocking still works fine, but I don't think lists get updated (last update was yesterday).

**Comment by translit** [comment]
>  all Adguard process takes 18OMb of RAM on my mac 

150Mb here. 47,700 rules. How much did uBO take?

**Comment by maciboy** [comment]
The guys at AdGuard are now thinking about how to circumvent the 50k limit using several blocker extensions 🎉 
https://github.com/AdguardTeam/AdGuardForSafari/issues/93

**Comment by translit** [comment]
> how to circumvent the 50k limit 

Good news. That limit is really a bummer.

**Comment by joshuaseltzer** [comment]
@translit I'm still using uBO on both Mojave and High Sierra (both on latest versions of Safari each).  Is there anyway to tell how much memory it uses?  It works great for the light browsing I do on these machines.

I might take a look at AdGuard though.  I'm using it for iOS and it works great.

**Comment by adrianmihalko** [comment]
Subscribe

**Comment by amarendra** [comment]
Maybe someone with repo access could try adding it to 
https://www.codeshelter.co

**Comment by brunnopleffken** [comment]
I tried AdGuard, it's nice but I feel it is still heavier than uBlock. It takes a few seconds longer until the main window and my favorites appear, which was not the case with uBlock.

**Comment by Beribis** [comment]
Can confirm AdGuard works, is noticeably slower but still effective. They on v1.21 and v1.4 will address >50k filters as other issues need to be addressed first. 
[v1.4 will increase filters at last - milestone](https://github.com/AdguardTeam/AdGuardForSafari/issues/93#issuecomment-464783214)

Sounds like they might like some help?
Thanks for the many many hours of enjoyment uBO has brought me and everyone I know. Will use it right up until they claw back Safari out of my hands. 

Let's hope Safari TP80 is implemented After AdGuard v1.4!

**Comment by strafe** [comment]
[Wipr](https://itunes.apple.com/it/app/wipr/id1320666476?l=en&mt=12) <sup>([or the free extension gallery version](https://safari-extensions.apple.com/details/?id=com.giorgiocalderolla.wipr-desktop-4449XA862Y))</sup>

**Comment by freediverx** [comment]
I've been using AdGuard for weeks and it's useless. Ridiculously slow and unresponsive, and totally ineffective at blocking ads and ad-blocker-blockers. Just try using it on the Washington Post site.

**Comment by ruchernchong** [comment]
@freediverx I do not see ads on Washington Post with AdGuard.

**Comment by freediverx** [comment]
Odd, now I can't reproduce it. 

Minutes ago, every Washington Post article was displaying an overlay telling me to disable my ad blocker. When I tried blocking the overlay, the underlying page loaded only partially and had scrolling disabled. Then after hitting refresh the overlay would appear again.

Now it seems to be working fine. (still laggy, but functional.)

**Comment by ruchernchong** [comment]
@freediverx are you using the browser extension or AdGuard for macOS (paid) or for iOS with AdGuard DNS (pro version)?

**Comment by freediverx** [comment]
Safari extension on macOS.

**Comment by ruchernchong** [comment]
You probably need the paid one to block ads on a system wide level.

**Comment by kusuriurikun** [comment]
> 
> 
> > There's no alternative that can do all that uBO does. So I will say it anyways -- Dump Safari, use Chrome **OR** Firefox on Mac and install uBO there.
> 
> Nah. Slow, no integration with Keychain, less shiny.

a) The only reason Safari is quicker is because (much like MSIE in the bad old days) part of the OS functionality is bound up in Safari.

b) Keychain login on macOS is arguably a security issue in and of itself (especially when used on a browser that is at this point essentially unsecurable from malware--and yes, kids, there are drive-by droppers for macOS malware).  That said, there are options out there to export the contents of an existing macOS keychain (including from iCloud) and use these with a modern, more robust, more provably secure password management system like KeePassX or Lastpass (with 99% of the keychain functionality you have with Safari now, up to and including autofilling passwords and being able to use master passwords, but also including autogeneration of secure passwords, portability to cloud storage (yes, including iCloud so should you desire), etc.  And yes, there are plugins for Chrome and Firefox (and browsers derived from them, including Opera and Waterfox) that allow quite seamless functionality similar to Safari and keychain management in macOS in general.

c) Frankly I can live without bling if it's a matter of actually _maintaining system security_--and with Safari effectively going full closed-garden this means at this point _the browser is impossible to properly secure against threats and thus should not be used for any purpose other than vendor-specific traffic or the download of a more  secure product_.  "Shiny" isn't going to help prevent system compromises.  I'll take "a wee bit not as shiny" for the basic functionality of being able to install systems that (among other things) prevent hijacks of browser and potentially the entire operating system via malicious scripts, prevent tracking (including "supercookies") by social media companies, prevent the misuse of my own computer resources via cryptocurrency miner scripts, oh, yeah, and block intrusive adverts as a benefit.

**Comment by ghost** [comment]
I hear you and I'm usually a Firefox enthusiast myself, using it exclusively since the Phoenix days, coming from Netscape, on all the systems I had trough the many years, until I got a MBP in 2016, as the impact on the battery life is noticeable (not as much with Nightly, it feels like). So I would welcome a UBO alternative for Safari (and overall, a more FF-like Browser experience, keeping the Safari UI, which I like), but we are past that it seems.

**Comment by freediverx** [comment]
Anyone suggesting replacing Safari with Chrome is hopefully clueless about security, privacy, and design.

**Comment by ghost** [comment]
To be fair (because you are quoting me), I didn't and I wouldn't do this at any point. I wouldn't advice anyone to use Chrome, and I would suggest to rather try Chromium if you have to (but I have no experience with either).

**Comment by kusuriurikun** [comment]
> 
> Anyone suggesting replacing Safari with Chrome is hopefully clueless about security, privacy, and design.
> […](#)
> On Aug 27, 2019, at 6:05 AM, chiffon-rouge ***@***.***> wrote: I hear you and I'm usually a Firefox enthusiast myself, using it exclusively since the Phoenix days, coming from Netscape, on all the systems I had trough the many years, until I got a MBP in 2016, as the impact on the battery life is noticeable (not as much with Nightly, it feels like). So I would welcome a UBO alternative for Safari (and overall, a more FF-like Browser experience, keeping the Safari UI, which I like), but we are past that it seems. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or mute the thread.

Personally, I'd actually be strongly recommending Waterfox (either v56 or the v68 alpha, which is pretty much Quantum-era Firefox + more privacy options and the ability to add at least some essential legacy extensions like DownThemAll), and re Chrome, there are multiple (MULTIPLE) Chromium forks if one does not wish to get pulled into the Google hydra.  

That said--even with all the data-scraping issues in Chrome--for now, you can at least install uBlock Origin and other datablocking apps for now (including HTML5 blocking apps), so it does have at least THAT advantage over Safari at the moment.  (Now, if Chrome goes on with its pigheaded ideas to sandbox things to the point adblockers no longer have the necessary access to actually adblock a la Safari, then Chrome can get consigned to the trashheap, too, and possibly the entire Chromium browser base if it turns out to be unfixable.  We aren't there yet, though, and it does seem that the main developers at le Google have backed off a bit...)

**Comment by joshuaseltzer** [comment]
So in September 2019, what is everyone using? On Mojave 10.14.6 I’m still using this extension, but it’s buggy and isn’t super reliable. From what I could see, it sounds like AdGuard and Wipr are options to look at. I’m looking for something that doesn’t hog on resources since my machines are slightly older.

**Comment by ruchernchong** [comment]
I am using AdGuard for Mac. It's great.

**Comment by joshuaseltzer** [comment]
@ruchern the full fledged app or the Safari Extension?

**Comment by ruchernchong** [comment]
@joshuaseltzer full-fledge app.

**Comment by Beribis** [comment]
> You can use [pihole](https://pi-hole.net) ⚫️ . It's much simpler solution and works for most ads in there. You don't even need to have installed plugin in your browser.
> 
> > Works as a ban list for specific domain address so ads are not loaded in your own browser via DNS.
> 
> Meaning pages suppose to load faster and you don't get those annoying message as "Please disable your ad block" etc. Also you can block easily ads on iOS this way (via wifi, cellular needs vpn)
> 
> Anyway here is public pihole server: 185.187.240.11
> Status data: https://pihole-uk.freek.ws/admin/
> 
> ### Setup
> On Mac your can set this server as your own DNS server via. Settings -> Network -> Advanced... -> DNS and there just add IP address `185.187.240.11` . They promise not using any logs.
> I have setup raspberry pi running own pihole.
> 
> Btw. I got that ip from here -> https://freek.ws/public-pi-hole/
> 
> Edit: Pihole is clever and interesting project but it's not perfect **yet** so it's recommended using combo pihole and 'some' adblock.

Firstly, thank you for posting this!
New site for Freek. Especially impressed by his intrusion protection and DNS amplification attack blocking.  I'm an instant fan ;)
 https://public-pihole.com

**Comment by frodo2007** [comment]
https://github.com/el1t/uBlock-Safari/issues/158#issuecomment-604391206

**Comment by frodo2007** [comment]
Explanation of the state of uBlock Origin (and other blockers) for Safari #158

**Comment by Beribis** [comment]
It’s over. Use Adguard for safari. Not to be confused with “adguard For macOS”. The first is free, unlike the second. Now works great, even with more than 50k records. Smooth replacement to uBO. Pihole, as mentioned before, does have stability issues from time to time. So much so I’ve stopped using it, not a problem with adguard plus on mobile you can use Malwarebytes

**Comment by translit** [comment]
> @Beribis Now works great, even with more than 50k records. 

No, it doesn't. AdGuard is slow.

> @Beribis  Pihole, as mentioned before, does have stability issues from time to time. 

No, it doesn't. Pi-Hole is stable as Debian.

**Comment by ifeeltiredboss** [comment]
> and use these with a modern, more robust, more provably secure password management system like KeePassX or **Lastpass**

@kusuriurikun Funny to read this condescending comment knowing of all the LastPass breaches

<img width="1281" alt="Screenshot 2023-01-15 at 11 40 54" src="https://user-images.githubusercontent.com/111905360/212536194-e9d4d340-aa8c-4236-be91-96911021f909.png">

**Comment by Beribis** [comment]
> > @Beribis Now works great, even with more than 50k records. 
> 
> No, it doesn't. AdGuard is slow.
> 
> > @Beribis  Pihole, as mentioned before, does have stability issues from time to time. 
> 
> No, it doesn't. Pi-Hole is stable as Debian.

Time to switch things up:
     Use nextdns.io, all blocklists are on the cloud. Nothing gets to your devices, updates are independent of any devices being switched on. Free tier works great for one device 🏆
